contrib/terraform/aws/create-infrastructure.tf

```diff
@@ -166,7 +166,7 @@ data "template_file" "inventory" {
     connection_strings_node   = join("\n", formatlist("%s ansible_host=%s", aws_instance.k8s-worker.*.private_dns, aws_instance.k8s-worker.*.private_ip))
     list_master               = join("\n", aws_instance.k8s-master.*.private_dns)
     list_node                 = join("\n", aws_instance.k8s-worker.*.private_dns)
-    connection_strings_etcd   = join("\n", formatlist("%s ansible_host=%s", ((var.aws_etcd_num > 0) ? (aws_instance.k8s-etcd.*.private_dns) : (aws_instance.k8s-master.*.private_dns)), ((var.aws_etcd_num > 0) ? (aws_instance.k8s-etcd.*.private_ip) : (aws_instance.k8s-master.*.private_ip))))
+    connection_strings_etcd   = join("\n", formatlist("%s ansible_host=%s", aws_instance.k8s-etcd.*.private_dns, aws_instance.k8s-etcd.*.private_ip))
```

Comment:
This reverts a part of https://github.com/kubernetes-sigs/kubespray/pull/7818

@faust64 Could you take a look at this change?

Reply:
That sounds wrong, indeed.

The idea was to allow for setting an etcd node count to 0.
If 0, we assume etcd would be collocated with the control plane, and generate our inventory having the etcd hostgroup mentioning master nodes.
Otherwise, etcd hostgroup should mention the corresponding instances

With that patch, if you set etcd count to 0, terreform would query aws for non-existing ec2 instances: your inventory might not generate at all, or best case, would be invalid.

There's nothing wrong in listing a given host twice, in your ansible inventory.
Arguably, having some `[etcd:children]` group, with your control plane group as a child group, should prevent you from duplicating 3 lines, if someone's up for that patch ...

May we ask again: what does this PR do? Concretely, what have you seen @rayterrill , that needs fixing?
I doubt you would be deploying clusters with some invalid inventory?

Sadly, I no longer have access to AWS API.
If my ternary somehow doesn't work: there should be an actual way to fix this.
For now, at best, that PR fixes inventory when using etcd dedicated nodes, while breaking its generation when collocating etcd with masters.

Reply:
Without this PR, it indeed doesn't appear to break anything, but it's just generating extra hosts in the inventory file which has been confusing for people reviewing PRs (especially those new to Ansible).

We're co-locating etcd with masters, and the inventory is fine (colocated masters is taken care of by the connection_strings_master entry, isn't it?). AFAIK - The only scenario that really should trigger anything printed here is the case where you're NOT colocating etcd hosts.

This just seemed like an extraneous entry - no need for the ternary operator here at all - thus the PR. If I'm mistaken, my fault.

Reply:
I'm not sure, you may be right.
When etcd_num is 0, don't you have an error resolving `aws_instance.k8s-etcd.*.private_dns`?
Or does that return you with an empty string?'

If there's no error, then I guess that change is OK then.

Reply:
We're running with etcd_num = 0 (colocated etcd), and no errors are generated during the generation of the Ansible inventory. I've made this PR's change locally and everything is fine - regenerated our clusters a number of times to test various scenarios, no issues any of those times.

Reply:
Fine by me, @oomichi , my mistake.

Reply:
Thanks for confirming @faust64